### ISSUE TITLE
[REF] Remove #type: ignore from SLM class

### ISSUE THREAD

**Issue by ReinderVosDeWael** [MICA-MNI/BrainStat#199]
Currently, mypy crashes on SLM.py due to an import inside a class. This issue is tracked at [mypy/issuse/10521](https://github.com/python/mypy/issues/10521).

**Comment by zihuaihuai** [comment]
mypy is temporary removed now, will mark this issue as add mypy back

**Comment by zihuaihuai** [comment]
Quick status check: the upstream mypy issue (python/mypy#10521) is still open. Several near-duplicates have been folded into it but no fix has landed — the in-class `from ._linear_model import _linear_model` style still trips mypy on current releases.

Concretely, removing line 1 of `SLM.py` and running `mypy brainstat/` against mypy 1.x reproduces a crash on the inner imports, so this issue is still blocked on the upstream resolution.

**Comment by zihuaihuai** [comment]
Resolved in #381. Mypy was crashing because of the in-class `from ._linear_model import _linear_model` pattern (python/mypy#10521). Lifting those imports to module level and binding them as class attributes (`_linear_model = _linear_model`) gets mypy past the crash, so the file-level `# type: ignore` shebangs are gone from both `SLM.py` and `_multiple_comparisons.py`. The ~380 pre-existing untyped-code errors that were hiding behind the blanket ignore now surface — those are a separate cleanup task; until they're triaged we're leaving the commented-out `mypy` CI job alone.
